server/tests/io/feed_parsers/belga_newsml_1_2_test.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013 - 2018 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+import os
+import datetime
+from lxml import etree
+from superdesk.tests import TestCase
+from belga.io.feed_parsers.belga_newsml_1_2 import BelgaNewsMLOneFeedParser
+
+
+class BelgaNewsMLOneTestCase(TestCase):
+    filename = 'belga_newsml_1_2.xml'
+
+    def setUp(self):
+        dirname = os.path.dirname(os.path.realpath(__file__))
+        fixture = os.path.normpath(os.path.join(dirname, '../fixtures', self.filename))
+        provider = {'name': 'test'}
+        with open(fixture, 'rb') as f:
+            parser = BelgaNewsMLOneFeedParser()
+            self.xml_root = etree.parse(f).getroot()
+            self.item = parser.parse(self.xml_root, provider)
+
+    def test_can_parse(self):
+        self.assertTrue(BelgaNewsMLOneFeedParser().can_parse(self.xml_root))
+
+    def test_content(self):
+        item = self.item[0]
+
+        self.assertEqual(
+            item['administrative']['foreign_id'],
+            'BIN118'
+        )
+        self.assertEqual(
+            item['administrative']['validation_date'],
+            '20190129T133356'
+        )
+        self.assertEqual(
+            item['administrative']['validator'],
+            'dwm'
+        )
+        self.assertEquals(item['authors'], [{'name': 'DWM', 'role': 'AUTHOR'}])
+        self.assertEqual(
+            item['body_html'],
+            '''Steven Van Geel gaf zich op 31 mei 2014 aan bij de politie van zijn thuisstad Leuven.'''
+        )
+        self.assertEqual(item['byline'], 'BELGA')
+        self.assertEqual(item['copyrightholder'], 'Belga')
+        self.assertEqual(item['date_id'], '20190129T133400')
+        self.assertEqual(item['extracity'], 'BRUSSEL')
+        self.assertEqual(item['extracountry'], 'België')
```

Comment:
if we are using vocabularies for this it should be stored as subject with scheme

Reply:
@petrjasek it's a custom text field. When I save an item via UI it stores it in 
```
"fields_meta" : {
		"extracountry" : {
			"draftjsState" : [
...
```
I've tried to put just text there, but it does not work, I see an empty textfield. This is something I'm trying to fix now.

Reply:
it should go into `extra` dict then imo, like:
```
item['extra'] = {'city': 'BRUSSEL', 'country': 'Belgie'}
```